src/utils/data-access.js

```diff
@@ -61,28 +61,31 @@ export async function syncSuggestions({
   // Remove outdated suggestions
   await Promise.all(
     existingSuggestions
-      .filter((existing) => !newDataKeys.has(buildKey(existing.data)))
+      .filter((existing) => !newDataKeys.has(buildKey(existing.getData())))
       .map((suggestion) => suggestion.remove()),
   );
 
   // Update existing suggestions
   await Promise.all(
     existingSuggestions
       .filter((existing) => {
```

Comment:
Is this filter redundant, since we already removed outdated suggestions above?

Reply:
`suggestion.remove` removes the entities from db, not from the `existingSuggestions` array; so those are have to be filtered again